### PR TITLE
Fix close event not getting creted

### PR DESCRIPTION
**Pull request by murillodigital**
The close user activity is not getting posted. When the call to post the event happens before stopAllScripts() it does get through. I looked at stopAllScripts and I can't identify what is being stopped/destroyed in that method that could affect UserActivityLogger::close(), the accountManager still exists and should be able to make an authenticatedrequest even after all scripts have been stopped.

 @ZappoMan I believe you're the last one to have worked on shutdown cleanup, I'm not sure if the best solution here is to move the close() useractivity up, move the script shutdown down (as its done in this PR) or altogether a different solution, considering I don't quite understand why the stopAllScripts call is what produces the problem.

### CODE REVIEW

**Comment by ZappoMan** [comment]
I don't see why stopping scripts earlier would cause your close event to not get sent... this change seems suspicious. Don't merge this till we can confirm that it doesn't introduce other problems.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 48de7cca3acc616fc04104c46baff71574851ed3 available at: https://gist.github.com/ef0ec51301d6a5b28c09<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4404.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4404.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -569,18 +568,20 @@ void Application::cleanupBeforeQuit() {
     _settingsThread.quit();
     saveSettings();
     _window->saveGeometry();
-    
+
     // TODO: now that this is in cleanupBeforeQuit do we really need it to stop and force
     // an event loop to send the packet?
     UserActivityLogger::getInstance().close();
 
+    ScriptEngine::stopAllScripts(this); // stop all currently running global scripts
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: we definitely want to make sure all scripts are stopped BEFORE we shut down the settings object... so we don't want this here.

**Comment by ZappoMan** [comment]
Ok, @murillodigital - I've been testing this repro case a little, and I can see what the problem is, but I don't think your change is actually going to fix this issue. I suspect this issue of "close" user acticvitiy has probably been broken for a while. At least since this code was added...

```
    // TODO: now that this is in cleanupBeforeQuit do we really need it to stop and force
    // an event loop to send the packet?
    UserActivityLogger::getInstance().close();
```

The problem is that we're not waiting for the network requests for the logging service to finish. Moving the scripts shutdown doesn't actually guarantee that we'll wait for the logging service... I think if it's working it's just "lucky" that it's working.

The right fix would be for the logging service and/or the AccountManager to keep track of it's pending requests... and have some mechanism to wait for some of them to finish.

**Comment by ZappoMan** [comment]
Closing this for a better fix.

**Pull request closed without merging**